lib/services/catalog/controllers/catalog-search-controller.js

```diff
@@ -12,20 +13,64 @@ class CatalogSearchController extends Controller {
 	}
 
 	get(req, res, next) {
-		const query = req.query.q;
+		const query = _.get(req, 'query.q', '');
 		const channel = (req.identity.channel || {}).id;
-		const type = req.query.type;
 
 		if (!channel) {
 			return next(Boom.badRequest('JSON Web Token must include a channel ID for search.'));
 		}
 
-		const args = {query, channel, type};
+		if (query.length <= 3) {
+			return next(Boom.badRequest('`q` query paramater must be at least 3 characters'));
+		}
+
+		let types;
+		if ((req.query || {}).types) {
+			// At this time, only allow sorting by one paramater.
+			types = req.query.types.split(',');
+		} else {
+			types = ['video', 'collection'];
+		}
+
+		let sortBy;
+		if ((req.query || {}).sort) {
+			// At this time, only allow sorting by one paramater.
+			sortBy = req.query.sort.split(',')[0];
+		} else {
+			sortBy = null;
+		}
+
+		const page = {
+			offset: parseInt(_.get(req, 'query.page.offset'), 10) || 0,
```

Comment:
Should these be:

`parseInt(_.get(req, 'query.page.offset', 0), 10)`
`parseInt(_.get(req, 'query.page.limit', 10), 10)`